public/gameInfo.php

```diff
@@ -89,6 +91,27 @@
 ?>
 <?php if ($gate): ?>
     <?php RenderContentStart($pageTitle) ?>
+    <script>
+        /**
+         * @param {number} currentPreferenceValue
+         * @param {number} gameId
+         */
+        function disableMatureContentWarningPreference(currentPreferenceValue, gameId) {
+            const newPreferencesValue = <?= $userDetails['websitePrefs'] | (1 << $matureContentPref) ?>;
```

Comment:
This either needs to check for the presence of `$userDetails`, or the entire function should be suppressed when a user is not logged in.

Reply:
You're right - should be fixed now.